### ISSUE TITLE
Need to resart ESP Box after few commands are recognized successfully

### ISSUE THREAD

**Issue by pk802** [toverainc/willow#262]
I am getting this error after few successful commands. 

![error](https://github.com/toverainc/willow/assets/99890240/d9aefc20-0c2f-494b-80fc-77daff6f96c6)
![error1](https://github.com/toverainc/willow/assets/99890240/11ee1842-b7b1-4c1e-b683-ad1b415b25be)

**Comment by kristiankielhofner** [comment]
The hosted server you are connecting to (infer.tovera.io) has served over 1,000 requests today with no logged connection errors according to metrics from the Cloudflare edge.

You are likely experiencing some connectivity issue(s) with your local network, internet connection, or somewhere in between.

This is going to be very difficult to diagnose because it involves some really fundamental things and a ton of variables.

**Comment by kristiankielhofner** [comment]
Are you using AMR-WB from the advanced settings?

**Comment by pk802** [comment]
Yes, I was using AMR-WB from the advanced setting. I changed it back to PCM. 
Having same issue. Works for 2 commands and then same error.

**Comment by pk802** [comment]
I tried https://wisng.tovera.io/api/willow?model=large&beam_size=5
as it was mentioned in few other issue and it seem to be working fine.

https://github.com/toverainc/willow/issues/200#issuecomment-1605240757

**Comment by nikito** [comment]
Have you tried running a continuous ping to the inference url to see if any drops are happening that could cause the socket to reset? Could leave it up and see if you are getting drops when the command fails. You can also run a continuous ping to the willow device to see if it is being dropped from WiFi for some reason; some APs will drop devices off they after before a minimum rssi.

Hope this helps! 🙂

**Comment by stintel** [comment]
Also which version are you running?

**Comment by kristiankielhofner** [comment]
> I tried https://wisng.tovera.io/api/willow?model=large&beam_size=5 as it was mentioned in few other issue and it seem to be working fine.
> 

This host is the exact some server (at a different hostname) with a different configuration for the speech recognition. The error you are getting comes from the Willow Inference Server when it fails to load the audio from your Willow device - before speech recognition configuration like model, beam_size, etc. None of these things should have any effect. I asked about AMR-WB because it's a different audio codec with additional steps in Willow and WIS to load the audio. It's under "Advanced Settings" because it should only be used in very special circumstances. We are working on updating the web interface to better explain the various settings.

As @nikito said your issues are almost certainly network related. We use a tried and true Wifi implementation from Espressif and HTTP/HTTPS (TCP) to transport the audio so it should arrive reliably to WIS even under challenging conditions. There is likely something specific to your environment that's especially challenging and causing the audio transfer to get corrupted, truncated, etc.

**Comment by pk802** [comment]
> Also which version are you running?

@stintel I am using latest code.



> Have you tried running a continuous ping to the inference url to see if any drops are happening that could cause the socket to reset? Could leave it up and see if you are getting drops when the command fails. You can also run a continuous ping to the willow device to see if it is being dropped from WiFi for some reason; some APs will drop devices if they are below a minimum rssi.
> 
> Hope this helps! 🙂

@nikito I ran continuous ping to inference URL and to my ESP32 box. There was no drop in packets. 

This problem still persist. Also I observed that after device is idle for few hours it dosen't work. I am sharing screenshot where you can see timestamp. I starts working again if I restart the ESP32 Box. 
![error5](https://github.com/toverainc/willow/assets/99890240/d5c01d31-7fc0-423a-ad5f-db6546e3b060)

**Comment by nikito** [comment]
Are you running the latest release of willow and WAS? I feel like this was reported a while back and solved. If so, it may be that something is wrong with this box, maybe it needs a full reflash? I've been running 3 boxes 24/7 and have not had any of them lock up or display an error like that running the latest rc1 build. 😊

**Comment by pk802** [comment]
> Are you running the latest release of willow and WAS? I feel like this was reported a while back and solved. If so, it may be that something is wrong with this box, maybe it needs a full reflash? I've been running 3 boxes 24/7 and have not had any of them lock up or display an error like that running the latest rc1 build. 😊

All my boxes were on rc1.  Now I have updated my 3 boxes to rc2 and WAS. Let me see if I get same errors.

**Comment by kristiankielhofner** [comment]
@pk802 - Have you tried again with our latest release?

**Comment by stintel** [comment]
We believe this is solved, and no feedback for > 1 year. Closing.
